### PR TITLE
Enable Kotlin

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR enables Kotlin for testing and migrates one test suite to this language.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol @dmdashenkov @yuri-sergiichuk 

Invited you all to review to show the syntax for tests in Kotlin. It's more condensed.

**Review by dmdashenkov — APPROVED** [review]
Looks good. We should also take a look at [spek](https://www.spekframework.org/running/) in future.

File: users-client/src/test/java/io/spine/net/NetChangeTest.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        tester.setDefault(EmailAddress::class.java, EmailAddress.getDefaultInstance())
+              .setDefault(Url::class.java, Url.getDefaultInstance())
+    }
+
+    @Nested
+    @DisplayName("not allow same")
+    inner class NotSame {
+
+        @Test
+        fun `email address`() {
```
Comment (armiol) [inline users-client/src/test/java/io/spine/net/NetChangeTest.kt:43]: Probably, `email addresses` — similar to `URL instances` down below.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov LGTM with a single suggestion.

**Review by yuri-sergiichuk — APPROVED** [review]
Looks promising. It'd be interesting to see how it maps to the BBC usage.

**Comment by yuri-sergiichuk** [comment]
> Looks good. We should also take a look at [spek](https://www.spekframework.org/running/) in future.

I personally don't like the approach tests style in Spek. It looks like a weirdish mix of JS Jasmine tests with Kotlin syntax over it. I'd better have it as in [Spock](http://spockframework.org/spock/docs/1.3/all_in_one.html#specifications_as_documentation).

**Pull request merged**